packages/core/src/openapi/openapi-request-builder.ts

```diff
@@ -47,13 +47,20 @@ export class OpenApiRequestBuilder<ResponseT = any> {
   async executeRaw(
     destination: Destination | DestinationNameAndJwt
   ): Promise<HttpResponse> {
-    return executeHttpRequest(destination, {
-      method: this.method,
-      url: this.getPath(),
-      headers: this.customHeaders,
-      params: this.parameters?.queryParameters,
-      data: this.parameters?.body
-    });
+    const fetchCsrfToken = ['post', 'put', 'patch', 'delete'].includes(
+      this.method.toLowerCase()
+    );
+    return executeHttpRequest(
+      destination,
+      {
+        method: this.method,
+        url: this.getPath(),
+        headers: this.customHeaders,
+        params: this.parameters?.queryParameters,
+        data: this.parameters?.body
+      },
+      { fetchCsrfToken }
+    );
```

Comment:
```suggestion
    return executeHttpRequest(
      destination,
      {
        method: this.method,
        url: this.getPath(),
        headers: this.customHeaders,
        params: this.parameters?.queryParameters,
        data: this.parameters?.body
      },
      { fetchCsrfToken: true }
    );
```
No, you don't have to consider the method, the `http-client` will ignore get.

Reply:
See [this](https://github.com/SAP/cloud-sdk-js/blob/main/packages/core/src/http-client/http-client.ts#L254) logic, which is used for deciding whether fetching the token is necessary.

Reply:
Hmm, then we might want to reconsider this logic... If I set this to true as a user, I would expect it to be done. However, I understand this might not make a lot of sense...
Independently, I would rather want to do it "right" here, as this is not obvious from outside... I will add a comment though, to remove it in v 2.0.

Reply:
You can do "right" here, since there might be some changes based on our discussion.

I'll write my thoughts here.
Currently, the OData request has a contract to handle the CSRF token automatically. However, there are also some hidden "optimisations", where the token request is skipped:
- when method is get (this logic has been moved from odata to generic http client)
- when `x-csrf-token` key exists in the header (this logic remains in the reused function, which is not touched)

In the long term, this option is set to `true` as default value, the user should care about the value and just delegate it to the SDK, which is exactly the same behaviour as the current OData. Otherwise, most of the users have to check the method and pass the right option (`true` of `false`) to the SDK.
So, "delegation" means the intelligent SDK will decide whether it needs to fetch the CSRF token for you. I'll document the option with details, since it's a public api.

I know this might solve 99% of the use cases and the corner case would be "my get request also need a csrf token".
So far, we never receive any feedback for this use case and the current SDK is not capable for this case anyways, I would not consider this case.
I would introduce a new flag `forceFetchCsrfToken` when we receive some requests from the users.

Reply:
I think `forceFetchCsrfToken` makes it more complicated than necessary, `fetchCsrfToken` should already be a "force" in my opinion.

Based on our discussion, I think we could also do the following (in version 2.0):
* `fetchCsrfToken` is not set, do the sdk default with optimizations
* `fetchCsrfToken` is false, don't fetch the csrf token ever
* `fetchCsrfToken` is true, fetch the csrf token, always